File: dataset_reduzido/codenet/Java/Code/s656827329.java
import java.util.*;
public class Main{
  public static void main(String[] args){
    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    int[] a = new int[n];
    for(int i = 0;i<n;i++){
      a[i] = sc.nextInt();
    }
    int count = 1;
    int now = 0;
    int bef = 0;
    for(int i = 1;i<n;i++){
      if(a[i-1]<a[i]){
        now = 1;
      }else if(a[i-1]>a[i]){
        now = -1;
      }else{
        continue;
      }
      if(bef == 0){
        bef = now;
      }
      if(bef != now){
        count++;
        bef = 0;
      }
    }
    System.out.println(count);
  }
}

File: output/LLaMa/codenet/Java/Python/s656827329.py
def main():
    n = int(input())
    a = [int(input()) for _ in range(n)]
    count = 1
    bef = 0
    for i in range(1, n):
        now = 0
        if a[i-1] < a[i]:
            now = 1
        elif a[i-1] > a[i]:
            now = -1
        if bef == 0:
            bef = now
        if bef != now:
            count += 1
            bef = 0
    print(count)

main()